client/trino-jdbc/src/main/java/io/trino/jdbc/TrinoConnection.java

```diff
@@ -119,7 +119,7 @@ public class TrinoConnection
         this.httpUri = uri.getHttpUri();
         uri.getSchema().ifPresent(schema::set);
         uri.getCatalog().ifPresent(catalog::set);
-        this.user = uri.getUser();
+        this.user = Optional.ofNullable(uri.getUser());
```

Comment:
Does a `null` username actually work. The CLI defaults the username to OS username but if you use the REST API a `X-Trino-User` is always needed.

Reply:
yep, it works :)